### ISSUE TITLE
Scrollspy not working with overflow: hidden on html

### ISSUE THREAD

**Issue by kubacode** [twbs/bootstrap#19565]
I'm trying to integrate a css only parallax with Scrollspy.

In order for the parallax effect to work correctly, overflow: hidden must be set on the html element. This causes the active scrollspy to get 'stuck' with the lowest element on the page.

I'm not able to create a jsfiddle, as adding the html { overflow: hidden } stops the fiddle from scrolling - doh! Is there a way to work around this to get scrollspy to work again, or should I move back to a JS based parallax?

Thanks in advance for your help!

**Comment by cvrebert** [comment]
What about a JS Bin instead of JS Fiddle?

**Comment by kubacode** [comment]
JSBin appears to have done the trick! http://jsbin.com/nifovag/edit?css,output

I haven't added the parallax, just the minimum viable CSS to replicate the issue. Let me know if you need the parallax added in too :)

**Comment by twbs-lmvtfy** [comment]
Hi @kubacode!

You appear to have posted a live example (http://jsbin.com/nifovag/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by kubacode** [comment]
Updated to http://jsbin.com/nifovag/edit?css,output to correct Bootlint error

**Comment by cvrebert** [comment]
@kubacode I'm unable to scroll your example in Chrome, except by clicking the navbar items. Is that intentional?

**Comment by kubacode** [comment]
@cvrebert Thats not intentional - should have mentioned that I'm using Safari - I'm experiencing my issue both on Mac and iOS versions. I'll obviously need to test further on other browsers.

**Comment by cvrebert** [comment]
@kubacode I'm unable to scroll your example in OS X Safari 9.1 or OS X Firefox 47 either.

**Comment by kubacode** [comment]
@cvrebert Dag nabit - I'm running Safari 8.0.7 - I'll update browsers and try to replicate the issue across Safari, Chrome & Firefox this evening - what version of Chrome are you running?

**Comment by cvrebert** [comment]
Chrome 49.0.2623.87 (latest stable)

**Comment by kubacode** [comment]
Ok, the JSBin should be updated with a working example in Safari 9.1 - for some reason the min-height no longer plays nice, so you may need to resize your window to be able to actually scroll.

The example also works in Firefox 47 & Chrome 49.0.2623.87, however the navbar stops being fixing to the top of the page in these browsers. This appears to be caused by the below CSS:
```
  -webkit-perspective: 1px;
  perspective: 1px;
  -webkit-transform-style: preserve-3d;
  transform-style: preserve-3d;
```
This is a separate issue which I don't think needs to be investigated as part of this issue.

Thanks for your patience and assistance by the way!

**Comment by cvrebert** [comment]
Removing the `perspective` and `transform-style`: http://output.jsbin.com/wohigo

**Comment by cvrebert** [comment]
@kubacode FYI, you're missing `position: relative`, which is required as explained in http://getbootstrap.com/javascript/#requires-relative-positioning
However, adding it didn't fix the problem :-/

**Comment by kubacode** [comment]
@cvrebert Touche! Upon further testing, adding an overflow, overflow-x or overflow-y attribute, with most properties (excluding visible, initial and inherit) to the html element causes the scrollspy to break.

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
